flow/network.cpp

```diff
@@ -178,6 +178,118 @@ std::string formatIpPort(const IPAddress& ip, uint16_t port) {
 	return format(patt, ip.toString().c_str(), port);
 }
 
+Optional<std::vector<NetworkAddress>> DNSCache::find(const std::string& host, const std::string& service) {
+	auto it = hostnameToAddresses.find(host + ":" + service);
+	if (it != hostnameToAddresses.end()) {
+		return it->second;
+	}
+	return {};
```

Comment:
Can you verify if this returns an `Optional<std::vector<NetworkAddress>>()` (not an optional with isPresent == true and empty list), since have our own optional class instead of std::optional?

Reply:
This is verified [here](https://github.com/apple/foundationdb/pull/6748/files#diff-ba692e310b2c4486b73998f9bfe19332320b2dde7560d910c54643bed519ed42R262).